### PR TITLE
fix(stonedb): fix build failed with c++1z.

### LINKED ISSUE #58: bug: build failed on c++1z
**Describe the problem**

1:Build the stonedb with build type - `realease`, and failed at sq/item_func.h. 
 error message:  string is not a member of 'std'. 
2: scoped_lock is available until c++17, so the build broke on `stondb/common/common_definitions.h`, which used `scoped_lock`.

**Expected behavior**

**How To Reproduce**

cmake ../../ -DCMAKE_BUILD_TYPE=Release 

**Environment**

**Additional context**

### PR DESCRIPTION
**Pull request by RingsC**
including:
      1:Removes the USE_CXX11 option from CMakeList, in fact, we use a lot of c++17 features in our code.
        therefore, the USE_CXX11 is not used anymore.
      2:Adds the missing header file.

Issue Number: close #58 

## Tests Check List

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Improvement
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Comment by hustjieke** [comment]
```
Issue Number: close https://github.com/stoneatom/stonedb/pull/59
```
I think the issue should be #58 ?

**Comment by RingsC** [comment]
> ```
> Issue Number: close https://github.com/stoneatom/stonedb/pull/59
> ```
> 
> I think the issue should be #58 ?

 the issue number updated.

File: CMakeLists.txt [hunk 1]
```diff
@@ -321,14 +321,21 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux" AND
     "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} -Wl,--build-id")
 ENDIF()
 
-OPTION(USE_CXX11 "Enable C++1z support (GCC only)" ON)
-IF (USE_CXX11 AND CMAKE_COMPILER_IS_GNUCC AND CMAKE_COMPILER_IS_GNUCXX)
+IF (CMAKE_COMPILER_IS_GNUCC AND CMAKE_COMPILER_IS_GNUCXX)
+  SET(CMAKE_C_FLAGS_RELEASE
+    "${CMAKE_C_FLAGS_RELWITHDEBINFO}")
```
Comment (hustjieke) [inline CMakeLists.txt]: `CMAKE_C_FLAGS_RELWITHDEBINFO` should be `CMAKE_C_FLAGS_RELEASE`

File: CMakeLists.txt [hunk 2]
```diff
@@ -321,14 +321,21 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux" AND
...
+  SET(CMAKE_CXX_FLAGS_RELEASE
+    "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -std=c++1z -fno-sized-deallocation")
```
Comment (hustjieke) [inline CMakeLists.txt]: `CMAKE_CXX_FLAGS_RELWITHDEBINFO` should be `CMAKE_CXX_FLAGS_RELEASE`

File: CMakeLists.txt [hunk 3]
```diff
@@ -321,14 +321,21 @@ IF(CMAKE_SYSTEM_NAME MATCHES "Linux" AND
...
+  SET(CMAKE_EXE_LINKER_FLAGS_RELEASE
+    "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} -static-libstdc++ -static-libgcc")
```
Comment (hustjieke) [inline CMakeLists.txt]: `CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO` should be `CMAKE_EXE_LINKER_FLAGS_RELEASE`

File: CMakeLists.txt (hunk 1 above)
Comment (RingsC) [reply CMakeLists.txt]: Yes, Thx.

File: CMakeLists.txt (hunk 2 above)
Comment (RingsC) [reply CMakeLists.txt]: Right, should be `CMAKE_CXX_FLAGS_RELEASE`

File: CMakeLists.txt (hunk 3 above)
Comment (RingsC) [reply CMakeLists.txt]: Yes,  Thx.

**Comment by hustjieke** [comment]
LGTM.

**Pull request merged**